models/game.py

```diff
@@ -0,0 +1,192 @@
+import struct
+import socket
+import numpy as np
+
+class Game:
+    """
+    Cette classe sert à intéragir avec le serveur, où se déroule le jeu.
+
+    Elle initie la connexion et update la map du terrain en fonction des messages qu'elle reçoit.
+
+    Paramètres
+    ----------
+    sock : socket
+        socket de connexion avec le serveur
+    shape : list
+        forme de la map
+    houses : list of tuples
+        localisation des maisons contenant les humains
+    depart : list
+        Coordonnées de départ
+    map : numpy array
+        Map du jeu (dimension shape x 3 - en 0 c'est le nombre d'humains, 1 le nombre de vampires et 2 le nombre de loups)
+    type : string
+        Type du joueur principal (vampire ou loup garous)
+    is_running : boolean
+        True si jeu est en cours False sinon
+    """
+
+    def __init__(self, hote="127.0.0.1", port=5555):
+        self._sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self._sock.connect((hote, port))
+        self._shape = None
+        self._houses = None
+        self._depart = None
+        self._map = None
+        self._type = None
+        self.is_running = True
+
+    def get_humans_positions(self):
+        """
+        Renvoie la position des humains sous forme de dictionnaires
+        (x, y) : nb_humans
+        avec (x, y) les coordonnées du groupe d'humains
+        et nb_humans le nombre d'humains
+        """
+        humans_dict = {}
+        for i, row in enumerate(self._map):
+            for j, cell in enumerate(row):
+                if cell[0] != 0:
+                    humans_dict[(i,j)] = cell[0]
+        print(humans_dict)
+        return humans_dict
+
+    def get_map(self):
+        return self._map
+
+    def get_opponent_positions(self):
```

Comment:
These methods are more related to the board than the game itself. I wonder if we could have a `Board` class with all the map related stuff and keep only the methods interacting with the server in the `Game`. Tell me what you think 🙂